src/Sharpliner.Model/GHActions/Webhook.cs

```diff
@@ -0,0 +1,450 @@
+            Closed,
+            Reopened,
+            Synchronize,
+            ReadyForReview,
+            Locked,
+            Unlocked,
+            ReviewRequested,
+            ReviewRequestRemoved,
+        }
+
+        /// <summary>
+        /// List of activities that will be considered when the event is triggered.
+        /// </summary>
+        public List<Activity> Activities { get; } = new();
+    }
+
+    /// <summary>
+    /// Triggers the workflow for the 'push' event.
+    /// </summary>
+    public record Push : Webhook;
+
+    /// <summary>
+    /// Triggers the workflow for the 'registry_package' event. For than one activity can trigger the
+    /// event. By default all activities trigger the workflow. Use the Activities property to filter them.
+    /// </summary>
+    public record RegistryPackage : Webhook
+    {
+
+        public enum Activity
+        {
+            Published,
+            Updated
+        }
+
+        /// <summary>
+        /// List of activities that will be considered when the event is triggered.
+        /// </summary>
+        public List<Activity> Activities { get; } = new();
+    }
+
+    /// <summary>
+    /// Triggers the workflow for the 'release' event. For than one activity can trigger the
+    /// event. By default all activities trigger the workflow. Use the Activities property to filter them.
+    /// </summary>
+    public record Release : Webhook
+    {
+        public enum Activity
+        {
+            Published,
+            Unpublished,
+            Created,
+            Edited,
+            Deleted,
+            Prereleased,
+            Released,
+        }
+
+        /// <summary>
+        /// List of activities that will be considered when the event is triggered.
+        /// </summary>
+        public List<Activity> Activities { get; } = new();
+    }
+    public record Status : Webhook {}
+
+    /// <summary>
+    /// Triggers the workflow for the 'watch' event. For than one activity can trigger the
+    /// event. By default all activities trigger the workflow. Use the Activities property to filter them.
+    /// </summary>
+    public record Watch : Webhook
+    {
+        public enum Activity
+        {
+            Started
+        }
+
+        /// <summary>
+        /// List of activities that will be considered when the event is triggered.
+        /// </summary>
+        public List<Activity> Activities { get; } = new();
+    }
+
+    /// <summary>
+    /// Triggers the workflow for the 'workflow_run' event. For than one activity can trigger the
+    /// event. By default all activities trigger the workflow. Use the Activities property to filter them.
+    /// </summary>
+    public record WorkflowRun : Webhook
+    {
+        public enum Activity
+        {
+            Completed,
+            Requested,
+        }
+
+        /// <summary>
+        /// List with the workflows that will trigger the event.
+        /// </summary>
+        [Required]
```

Comment:
Where does this attribute come from?

Reply:
From the BCL but we might want to add our own in the near future. Using it atm but maybe removing it might be a good idea.